core-lib/Kernel.ns

```diff
@@ -64,7 +64,7 @@ class Kernel vmMirror: vmMirror = Object <: Value (
     public notNil = ( ^ true )
     public ifNil: aBlock = ( ^ self )
     public ifNotNil: aBlock = ( ^ aBlock value )
-    public ifNil: noGoBlock ifNotNil: goBlock = ( ^ goBlock value: self )
```

Comment:
This was actually correct. But `ifNotNil:` seems wrong.
The reason is that we want to capture the value and have it in the block argument:

```Smalltalk
obj ifNotNil: [:it | it isGuaranteedToExist ].
(* or also *)
obj ifNil: [ #foo ] ifNotNil: [:it <SomeThing> | it baz ].
```

This is used for instance in MiniTest.

Reply:
More digging gives both uses, with, and without `:it`.
Think we need `cull:` then to give a missing value as `nil`.

Reply:
Something to consider: http://forum.world.st/cull-protocol-td1560122.html

Reply:
What is your preference for implementing this: Kernel, Primitive, or both?

Reply:
First in Kernel, and if it turns out to be a performance concern, we can optimize it?

Reply:
Sure, we need an numArgs primitive for that anyway, do we want to expose that for other use-cases or just culling?

Reply:
Good question. We do not expose it in Symbol either, and I'd consider it reflective information.
So strictly speaking it should be only exposed via mirrors.
Think the best would be to simply use a `vmMirror` primitive directly in the `cull*` implementations.